GrandCentralBoard/Widgets/Harvest/DailyBillingStats.swift

```diff
@@ -11,36 +11,35 @@ import Decodable
 
 struct DailyBillingStats {
     let day: NSDate
-    let groups: [BillingStatsGroup]
+    let billings: [DayEntry]
 }
 
+struct DayEntry: Decodable {
+    let userID: Int
+    let hours: Double
 
-extension DailyBillingStats {
-    static func emptyStats(day: NSDate = NSDate()) -> DailyBillingStats {
-        return DailyBillingStats(day: day, groups: BillingStatsGroup.allTypes().map { BillingStatsGroup(type: $0, count: 0, averageHours: 0) })
+    static func decode(json: AnyObject) throws -> DayEntry {
+        return try DayEntry(userID: json => "user_id", hours: json => "hours")
     }
+}
 
-    func merge(dailyBillingStats: DailyBillingStats) -> DailyBillingStats {
-        let types = BillingStatsGroup.allTypes()
-        let indexes = 0..<types.count
+private struct Day: Decodable {
+    let date: NSDate
 
-        let groups = indexes.map { index -> BillingStatsGroup in
-            let group1 = self.groups[index]
-            let group2 = dailyBillingStats.groups[index]
-            let count = group1.count + group2.count
-            let totalHours = group1.averageHours * Double(group1.count) + group2.averageHours * Double(group2.count)
-            let averageHours = totalHours / max(1.0, Double(count))
-            let type = types[index]
+    static func decode(json: AnyObject) throws -> Day {
+        let formatter = NSDateFormatter()
```

Comment:
`NSDateFormatters` are expensive. Once created, this it should be cached for later usage, since `Day` struct is decoded many times for this widget.